src/shaders/hillshade.fragment.glsl

```diff
@@ -26,10 +27,23 @@ void main() {
     float azimuth = u_light.y + PI;
     float polar = u_light.z;
 
+    // We scale the slope exponentially based on intensity, using a calculation similar to 
+    // the exponential interpolation function in the style spec: 
+    // https://github.com/mapbox/mapbox-gl-js/blob/master/src/style-spec/expression/definitions/interpolate.js#L217-L228
+    // so that higher intensity values create more opaque hillshading.
+    slope = intensity != 0.5 ? 
+        ((pow(1.875 - intensity * 1.75, slope) - 1.0) / (pow(1.875 - intensity * 1.75, (0.5 * PI)) - 1.0)) * (0.5 * PI) 
+        : slope; 
+
+    // The accent color is calculated with the cosine of the slope while the shade color is calculated with the sine
+    // so that the accent color's rate of change eases in while the shade color's eases out.
     float accent = cos(slope);
-    vec4 accent_color = intensity * clamp((1.0 - accent) * 2.0, 0.0, 1.0) * u_accent;
+    // We multiply both the accent and shade color by a clamped intensity value
+    // so that intensities >= 0.5 do not additionally affect the color values
+    // while intensity values < 0.5 make the overall color more transparent.
+    vec4 accent_color = (1.0 - accent) * u_accent * clamp(intensity * 2.0, 0.0, 1.0);
     float shade = abs(mod((aspect + azimuth) / PI + 0.5, 2.0) - 1.0);
-    vec4 shade_color = mix(u_shadow, u_highlight, shade) * slope * sin(polar) * mix(0.0, 2.0, intensity);
```

Comment:
if we're not using the polar angle anymore for rendering, we should remove it (L26)